apps/igloo-kit-cli/src/cli/webserver.rs

```diff
@@ -0,0 +1,68 @@
+use hyper::Body;
+use hyper::Request;
+use hyper::Response;
+use hyper::Server;
+use hyper::StatusCode;
+use hyper::service::service_fn;
+use std::convert::Infallible;
+use std::path::PathBuf;
+use hyper::service::make_service_fn;
+use super::Message;
+use super::MessageType;
+use super::user_messages::show_message;
+use std::collections::HashSet;
+use std::sync::Mutex;
+use std::sync::Arc;
+use super::CommandTerminal;
+
+
+async fn handler(req: Request<Body>, route_table: Arc<Mutex<HashSet<PathBuf>>>) -> Result<Response<String>, Infallible> {
+    let route_prefix = PathBuf::from("/");
+    let route = PathBuf::from(req.uri().path()).strip_prefix(route_prefix).unwrap().to_path_buf();
+
+    // Check if route is in the route table
+    if route_table.lock().unwrap().contains(&route) {
```

Comment:
You have to lock the implementation to read? Does it mean you can't write concurrently?

Reply:
With a mutex, that is correct. You have to lock to read. We could try a RWLock to see if we can get concurrent readers. I had concerns about the ability to pass the RWLock to the thread which is why I didn't use it. I'll add a comment to investigate the RWLock but we can probably differ that to future releases